### PR TITLE
[routing-manager] handle deprecated OMR prefix

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit enhances routing-manager to avoid using deprecated (i.e. P_preferred=false) OMR prefixes as the winning OMR prefix, but still sends RA RIO for the deprecating OMR prefixes.

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. Thanks.

File: src/core/border_router/routing_manager.cpp [hunk 1]
```diff
@@ -447,9 +450,9 @@ void RoutingManager::EvaluateOmrPrefix(OmrPrefixArray &aNewOmrPrefixes)
 
     // Decide if we need to add or remove our local OMR prefix.
 
-    if (aNewOmrPrefixes.IsEmpty())
+    if (electedOmrPrefix == nullptr)
     {
-        LogInfo("EvaluateOmrPrefix: No valid OMR prefixes found in Thread network");
+        LogInfo("EvaluateOmrPrefix: No non-deprecating OMR prefixes found in Thread network");
```
Comment (wgtdkp) [inline src/core/border_router/routing_manager.cpp:455]: nit:
```suggestion
        LogInfo("EvaluateOmrPrefix: No preferred OMR prefixes found in Thread network");
```

File: tests/scripts/thread-cert/border_router/test_manual_omr_prefix.py [hunk 2]
```diff
@@ -87,6 +89,28 @@ def test(self):
...
+
+        BR_1_OMR_PREFIX = pv.vars['BR_1_OMR_PREFIX']
+        assert BR_1_OMR_PREFIX.endswith('::/64')
+        BR_1_OMR_PREFIX = BR_1_OMR_PREFIX[:-3]
+
+        # Filter all ICMPv6 RA messages advertised by BR_1
+        ra_pkts = pkts.filter_eth_src(BR_1_ETH).filter_icmpv6_nd_ra()
+
+        # Verify BR_1 sends RA RIO for both BR OMR prefix and deprecating OMR prefix (i.e. 2004::/64)
+        ra_pkts.filter(lambda p: ICMPV6_RA_OPT_TYPE_RIO in p.icmpv6.opt.type and '2004::' in p.icmpv6.opt.prefix and
```
Comment (wgtdkp) [inline tests/scripts/thread-cert/border_router/test_manual_omr_prefix.py:112]: There are multiple test cases in this class, not sure if they could interfere each other?

**Review by wgtdkp — APPROVED** [review]
LGTM 👍 Thanks!

File: tests/scripts/thread-cert/border_router/test_manual_omr_prefix.py (hunk 2 above)
Comment (simonlingoogle) [reply tests/scripts/thread-cert/border_router/test_manual_omr_prefix.py:112]: They won't . Note that the prefix is always removed at the end of each subtest.

**Review by jwhui — APPROVED** [review]

**Pull request merged**